packages/logger/src/transports/storage.ts

```diff
@@ -307,38 +314,15 @@ export class StorageTransport implements ITransport {
     return this._options.maxLogsSize ?? DEFAULT_MAX_LOGS_SIZE;
   }
 
-  // fast way to get total size of logs
-  protected async _getTotalSize() {
-    const logs = (await this._table?.toArray()) ?? [];
-    const totalLogSize = logs.reduce((acc, log) => {
-      return acc + (log.size || 0);
-    }, 0);
-    return totalLogSize;
-  }
-
   protected async _pruneLogs() {
     await this._deleteExpiredLogs();
-    // only prune logs if total size is greater than maxLogsSize
-    const totalLogSize = await this._getTotalSize();
-    if (totalLogSize > this.maxLogsSize) {
-      let sizeOverBy = this.maxLogsSize;
-      let cutoffTime = 0;
-
-      await this._table
-        ?.orderBy('time')
-        .reverse()
-        .until((log: Logs) => {
-          sizeOverBy -= log.size;
-          if (sizeOverBy <= 0) {
-            cutoffTime = log.time;
-            return true;
-          }
-          return false;
-        })
-        .toArray();
-
-      if (cutoffTime > 0) {
-        await this._deleteLogs(cutoffTime);
+    const logs = (await this._getLogs()) ?? [];
+    let totalSize = 0;
+    for (const log of logs) {
+      totalSize += log.size;
+      if (totalSize > this.maxLogsSize) {
+        await this._deleteLogs(log.time);
```

Comment:
if the id have order logic, how about use some SQL to delete

like DELETE XXX  WHNE value > XXX